src/launchpad/artifacts/android/apk.py

```diff
@@ -114,20 +117,34 @@ def get_app_icon(self) -> bytes | None:
             logger.info("No application element found in manifest")
             return None
 
-        icon_path = manifest.application.icon_path
-        if not icon_path:
+        icon_path_str = manifest.application.icon_path
+        if not icon_path_str:
             logger.info("No icon path found in manifest")
             return None
 
-        icon_path = self._extract_dir / icon_path
+        icon_path = self._extract_dir / icon_path_str
 
         if not icon_path.exists():
+            logger.info(f"Icon not found in APK: {icon_path_str}")
             return None
 
-        # TODO(EME-461): Support XML icon paths
+        # Handle XML drawables (adaptive icons, vector drawables, shapes)
         if icon_path.suffix == ".xml":
-            logger.info(f"Icon path {icon_path} is an XML file, which is not yet supported. Skipping.")
-            return None
+            try:
+                binary_res_tables = self.get_resource_tables()
+
+                binary_xml_drawable_utils = BinaryXmlDrawableParser(self._extract_dir, binary_res_tables)
+
+                icon = binary_xml_drawable_utils.render_from_path(icon_path)
```

Comment:
I recommend adding a test case for `get_app_icon ()` like I did for the iOS app icon parsing which basically just checks whether this creates a valid PNG image from our HackerNews fixture.

Reply:
Will do, good idea